core/tests/puppeteer-acceptance-tests/user-utilities/question-admin-utils.js

```diff
@@ -0,0 +1,163 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Question admin users utility file.
+ */
+
+const baseUser = require(
+  '../puppeteer-testing-utilities/puppeteer-utils.js');
+const testConstants = require(
+  '../puppeteer-testing-utilities/test-constants.js');
+const contributorDashboardAdminUrl =
+  testConstants.URLs.ContributorDashboardAdmin;
+const { showMessage } = require(
+  '../puppeteer-testing-utilities/show-message-utils.js');
+
+const reviewQuestionRightValue = 'question';
+const submitQuestionRightValue = 'submit_question';
+const usernameMethodValue = 'username';
+const roleMethodValue = 'role';
+
+// "View Contributor Dashboard Users" form elements.
+const viewContributorFilterMethodSelect =
+  'select#view-contributor-filter-method-select';
+const viewContributerUsernameInput =
+  'input#view-contributor-username-input';
+const viewContributorCategorySelect =
+  'select#view-contributor-category-select';
+const viewContributorSubmitButton =
+  'button#view-contributor-submit-button';
+
+const viewContributorReviewQuestionsResult =
+  '.e2e-test-question-reviewer';
+const viewReviewQuestionRoleUserResult =
+  '.e2e-test-reviewer-roles-result';
+
+// "Add Contribution Rights" form elements.
+const addContributorUsernameInput =
+  'input#add-contribution-rights-user-input';
+const addContributonRightsCategorySelect =
+  'select#add-contribution-rights-category-select';
+const addContributionRightsSubmitButton =
+   'button#add-contribution-rights-submit-button';
+
+module.exports = class QuestionAdmin extends baseUser {
+  /**
+   * Function for navigating to the contributor dashboard admin page.
+   */
+  async navigateToContributorDashboardAdminPage() {
+    await this.goto(contributorDashboardAdminUrl);
+  }
+
+  /**
+   * Function for adding a right of reviewing questions to a user.
+   * @param {string} username - the username of the user.
+   */
+  async addReviewQuestionRights(username) {
+    await this.type(addContributorUsernameInput, username);
+    await this.select(
+      addContributonRightsCategorySelect, reviewQuestionRightValue);
+    await this.clickOn(addContributionRightsSubmitButton);
+
+    await this.page.waitForNetworkIdle();
+  }
+
+  /**
+   * Function for adding a right of submitting questions to a user.
+   * @param {string} username - the username of the user.
+   */
+  async addSubmitQuestionRights(username) {
+    await this.type(addContributorUsernameInput, username);
+    await this.select(
+      addContributonRightsCategorySelect, submitQuestionRightValue);
+    await this.clickOn(addContributionRightsSubmitButton);
+
+    await this.page.waitForNetworkIdle();
+  }
+
+  /**
+   * Function to display contribution rights by user.
+   * @param {string} username - the username of the user to view.
+   */
+  async viewContributionRightsForUser(username) {
```

Comment:
This function "does nothing", I don't think it should be exposed. If anything it might be a private function that other functions in this file make use of.

Reply:
Done